### DISCUSSION TITLE
Podman fuse-overlayfs

### DISCUSSION THREAD

**Discussion by jesseborden** [containers/podman#21962] [General]
Based on this:

https://github.com/containers/podman/blob/main/docs/tutorials/performance.md 

It appears the RHEL 8 kernel should not be using native overlayfs as the kernel is not >= 5.12.

However, if you run the command to check the overlay diff without changing the storage.conf for the non-root user, it will show true for:

podman info -f ‘{{index .Storage.GraphStatus “Native Overlay Diff”}}

This does not seem right if the non-root user should by default not be using native overlayfs and no changes were made to the default podman install. It is almost as if the default install thinks the non-root user should be using native overlayfs.

But, If I set the storage.conf for the user specifically to fuse-overlays, the result of the podman info command above looks more correct and returns false indicating it is using fuse-overlayfs. Changing the options globally seems to have no impact on the non-root users result when running podman info command even when no storage.conf exists for the given user.

We have had a lot of issues with COPY commands in dockerfiles being extremely slow and hoping this change for the user will help, but the performance.md page related to the output of podman info command seems strange with regards to specified kernel support and default behavior.

**Comment by eriksjolund** [comment]
Quote from [Kernel Newbies summary about  Linux 5.11](https://kernelnewbies.org/Linux_5.11#Unprivileged_Overlayfs_mounts)
_this release finally allows non-privileged users to make use of Overlayfs._






Quote from  [Red Hat Enterprise Linux 8.5 release notes](https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/8/html-single/8.5_release_notes/index)
_Native overlay file system support in the kernel is now available.
The overlay file system support is now available from kernel 5.11.
The non-root users will have native overlay performance even when running rootless (as a user)_

It looks like a documentation bug. I guess it should be _5.11_ instead of _5.12_ in this line
https://github.com/containers/podman/blob/56e0f063bd280318a861d1c7f816160c31fdf7bb/docs/tutorials/performance.md?plain=1#L86



(I don't remember where I got the version number _5.12_ from when I wrote the text https://github.com/containers/podman/commit/b5ee4de8cf3a05a56f32f4a739fcac24ba4023fd)

**Comment by rhatdan** [comment]
Please open a PR to fix the documentation to state that RHEL 8 has support.

**Reply by jesseborden** [reply]
I opened an issue and referenced this. I'm not pro-rated in podman yet to open that PR. Our builds went from 1 hour to 3 hours from RHEL 7 docker, to RHEL 8 podman, so I'm just turning over rocks to try to improve performance.

**Reply by rhatdan** [reply]
Unfair comparison since you went from rootful to rootless.  If you ran podman rootful, it should perform similarly.  Anyways if you just use "overlay" in rootless mode, it should get similar performance without using fuse-overlayfs
